tests/l2/test_l2_configure.py

```diff
@@ -0,0 +1,91 @@
+"""
+Tests related to L2 configuration
+"""
+import logging
+import pytest
+import time
+
+from tests.common import reboot
+from tests.common.helpers.assertions import pytest_expect
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+
+def test_l2_configure(duthosts, rand_one_dut_hostname):
+    """
+    @summary: Test configuring dut as a L2 switch.
+
+    Args:
+        duthosts: set of DUTs.
+    """
+    # Setup.
+    duthost = duthosts[rand_one_dut_hostname]
+    hwsku = duthost.facts["hwsku"]
+
+    # Store original config for comparison.
+    orig_vlan = duthost.shell("show vlan config")["stdout"]
+    orig_int = duthost.shell("show int status")["stdout"]
+
+    # Perform L2 configuration
+    logger.info("Configuring DUT into L2 mode.")
+    l2_cfg = "sudo sonic-cfggen --preset l2 -p -H -k {}" \
+        " | sudo config load /dev/stdin -y".format(hwsku)
+    duthost.shell(l2_cfg)
+    duthost.shell("sudo config qos reload --no-dynamic-buffer")
+    time.sleep(30)
```

Comment:
>time.sleep(30) [](http://example.com/codeflow?start=4&length=14)

We should try best to remove sleeping for fixed time:
1. if too long, waste overall test runtime
2. if too short, test may be flaky on some weaker hwsku

Could you use function "wait_until" for a special signal? You may use `redis-cli monitor` to observe what is really happening.

Or maybe you do not need to sleep at all?

Reply:
I removed this. This is replaced by a reboot (which is one step in the expected test case).